docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_integrated_cloudwatch_remediation.mdx

```diff
@@ -1,6 +1,187 @@
+4. If the "Log group" and "Log role ARN" fields are populated, it means that CloudTrail events are being monitored by CloudWatch Logs. If these fields are not populated, it indicates a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. **Check the existing CloudTrail trails:**
+   First, you need to check the existing CloudTrail trails. You can do this by using the AWS CLI command `describe-trails`. This command will return a list of trail ARNs (Amazon Resource Names) and their respective settings.
+
+   Command:
+   ```
+   aws cloudtrail describe-trails
+   ```
+
+2. **Check the CloudWatch Logs for each trail:**
+   For each trail returned by the previous command, you need to check if it has a CloudWatch Logs log group associated with it. This can be done by checking the `CloudWatchLogsLogGroupArn` field in the trail's settings.
+
+   You can use the `get-trail-status` command to get the status of a specific trail. Replace `<TrailName>` with the name of the trail you want to check.
+
+   Command:
+   ```
+   aws cloudtrail get-trail-status --name <TrailName>
+   ```
+
+3. **Check the CloudWatch Logs log group:**
+   If a trail has a CloudWatch Logs log group associated with it, you need to check if the log group is receiving log events from CloudTrail. This can be done by checking the `lastEventReceived` field in the log group's settings.
+
+   You can use the `describe-log-groups` command to get the settings of a specific log group. Replace `<LogGroupName>` with the name of the log group you want to check.
+
+   Command:
+   ```
+   aws logs describe-log-groups --log-group-name <LogGroupName>
+   ```
+
+4. **Check the CloudWatch Logs metric filters:**
+   Finally, you need to check if the log group has any metric filters that match CloudTrail events. This can be done by checking the `filterPattern` field in the metric filter's settings.
+
+   You can use the `describe-metric-filters` command to get the metric filters of a specific log group. Replace `<LogGroupName>` with the name of the log group you want to check.
+
+   Command:
+   ```
+   aws logs describe-metric-filters --log-group-name <LogGroupName>
+   ```
+   
+If any of these checks fail, it means that CloudTrail events are not being monitored by CloudWatch Logs.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: To start with, you need to import the necessary libraries in your Python script. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of AWS services like Amazon S3, Amazon EC2, etc. 
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+Then, establish a session using your AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>